engine/cmd/pull.go

```diff
@@ -33,26 +41,121 @@ func pullCmd() *cobra.Command {
 				return nil
 			}
 
+			if runOnce {
+				if o.modelID == "" {
+					return fmt.Errorf("model ID must be set on the run-once mode")
+				}
+			} else {
+				if socketPath == "" {
+					return fmt.Errorf("socket path must be set on the daemon mode")
+				}
+				if o.runtime != config.RuntimeNameOllama {
+					return fmt.Errorf("daemon mode is only available for the ollama")
+				}
+			}
+
 			c, err := config.Parse(path)
 			if err != nil {
 				return err
 			}
 
-			return pull(cmd.Context(), o, c)
+			ctx, cancel := signal.NotifyContext(cmd.Context(), syscall.SIGTERM)
+			defer cancel()
+			if runOnce {
+				return pull(ctx, o, c)
+			}
+			return runServer(ctx, c, socketPath)
 		},
 	}
 	cmd.Flags().IntVar(&o.index, "index", 0, "Index of the pod")
 	cmd.Flags().StringVar(&o.runtime, "runtime", "", "Runtime name for the model")
 	cmd.Flags().StringVar(&o.modelID, "model-id", "", "Model ID to be registered")
 	cmd.Flags().StringVar(&path, "config", "", "Path to the config file")
 	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Pull the model even if its index is not 0")
+	cmd.Flags().BoolVar(&runOnce, "run-once", true, "Run the command only once (daemon mode is only available for the ollama model)")
```

Comment:
No strong opinion, so please feel free to ignore, but slightly prefer defining a flag such that the default value is `false` (e.g., `--enable-daemon-mode`).